### PR TITLE
Prevent the cross-encoder logic from being applied to classification tasks

### LINKED ISSUE #18727: [Bug]: OpenAI Classification Client returning logits instead of softmax values
<details>
  <summary>environment info</summary>

```
### Your current environment

==============================
        System Info
==============================
OS                           : Ubuntu 22.04.4 LTS (x86_64)
GCC version                  : (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version                : Could not collect
CMake version                : version 3.22.1
Libc version                 : glibc-2.35

==============================
       PyTorch Info
==============================
PyTorch version              : 2.7.0+cu126
Is debug build               : False
CUDA used to build PyTorch   : 12.6
ROCM used to build PyTorch   : N/A

==============================
      Python Environment
==============================
Python version               : 3.10.12 (main, Feb  4 2025, 14:57:36) [GCC 11.4.0] (64-bit runtime)
Python platform              : Linux-5.15.0-139-generic-x86_64-with-glibc2.35

==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : 11.5.119
CUDA_MODULE_LOADING set to   : LAZY
GPU models and configuration : 
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090

Nvidia driver version        : 535.230.02
cuDNN version                : Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.9.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv.so.9.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn.so.9.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_precompiled.so.9.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_runtime_compiled.so.9.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_graph.so.9.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_heuristic.so.9.2.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops.so.9.2.1
HIP runtime version          : N/A
MIOpen runtime version       : N/A
Is XNNPACK available         : True

==============================
          CPU Info
==============================
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        48 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               24
On-line CPU(s) list:                  0-23
Vendor ID:                            AuthenticAMD
Model name:                           AMD Ryzen 9 5900X 12-Core Processor
CPU family:                           25
Model:                                33
Thread(s) per core:                   2
Core(s) per socket:                   12
Socket(s):                            1
Stepping:                             0
Frequency boost:                      enabled
CPU max MHz:                          3700.0000
CPU min MHz:                          2200.0000
BogoMIPS:                             7385.88
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm
Virtualization:                       AMD-V
L1d cache:                            384 KiB (12 instances)
L1i cache:                            384 KiB (12 instances)
L2 cache:                             6 MiB (12 instances)
L3 cache:                             64 MiB (2 instances)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-23
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Mitigation; safe RET, no microcode
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Retpolines; IBPB conditional; IBRS_FW; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

==============================
Versions of relevant libraries
==============================
[pip3] numpy==2.2.6
[pip3] nvidia-cublas-cu12==12.6.4.1
[pip3] nvidia-cuda-cupti-cu12==12.6.80
[pip3] nvidia-cuda-nvrtc-cu12==12.6.77
[pip3] nvidia-cuda-runtime-cu12==12.6.77
[pip3] nvidia-cudnn-cu12==9.5.1.17
[pip3] nvidia-cufft-cu12==11.3.0.4
[pip3] nvidia-cufile-cu12==1.11.1.6
[pip3] nvidia-curand-cu12==10.3.7.77
[pip3] nvidia-cusolver-cu12==11.7.1.2
[pip3] nvidia-cusparse-cu12==12.5.4.2
[pip3] nvidia-cusparselt-cu12==0.6.3
[pip3] nvidia-nccl-cu12==2.26.2
[pip3] nvidia-nvjitlink-cu12==12.6.85
[pip3] nvidia-nvtx-cu12==12.6.77
[pip3] pyzmq==26.4.0
[pip3] torch==2.7.0
[pip3] torchaudio==2.7.0
[pip3] torchvision==0.22.0
[pip3] transformers==4.52.3
[pip3] triton==3.3.0
[conda] Could not collect

==============================
         vLLM Info
==============================
ROCM Version                 : Could not collect
Neuron SDK Version           : N/A
vLLM Version                 : 0.9.0
vLLM Build Flags:
  CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
  	GPU0	GPU1	CPU Affinity	NUMA Affinity	GPU NUMA ID
GPU0	 X 	PHB	0-23	0		N/A
GPU1	PHB	 X 	0-23	0		N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

==============================
     Environment Variables
==============================
NCCL_CUMEM_ENABLE=0
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1
CUDA_MODULE_LOADING=LAZY
```
</details>

### 🐛 Describe the bug

After installing `0.9.0` pre-release from the following wheel https://wheels.vllm.ai/c533c6fa8403f8bb3a75f130e063dbaafc1d69dc/vllm-0.9.0-cp38-abi3-manylinux1_x86_64.whl, I see that for `papluca/xlm-roberta-base-language-detection` model raw logits are being returned instead of softmax output.

#### Steps to reproduce

```
pip install https://wheels.vllm.ai/c533c6fa8403f8bb3a75f130e063dbaafc1d69dc/vllm-0.9.0-cp38-abi3-manylinux1_x86_64.whl

vllm serve 'papluca/xlm-roberta-base-language-detection' --port 8098 --host 0.0.0.0 --task classify
```

#### Wrong output

When calling the model, I see raw logits being returned:

```
curl -v "http://127.0.0.1:8098/classify" \
  -H "Content-Type: application/json" \
  -d '{
    "model": "papluca/xlm-roberta-base-language-detection",
    "input": "Hello"
  }'
```

Returns:

```json
{
  "id": "classify-5b23c5f8efa24ccfb1ca60f6b4283006",
  "object": "list",
  "created": 1748283054,
  "model": "papluca/xlm-roberta-base-language-detection",
  "data": [
    {
      "index": 0,
      "label": "en",
      "probs": [
        -1.0625,
        0.290771484375,
        -1.125,
        -0.9794921875,
        -0.339111328125,
        -0.1148681640625,
        -0.75439453125,
        0.360107421875,
        -0.479736328125,
        1.2421875,
        -0.338623046875,
        0.671875,
        -0.04315185546875,
        3.888671875,
        -0.52099609375,
        -0.26416015625,
        0.2705078125,
        -0.152099609375,
        0.71484375,
        -0.5517578125
      ],
      "num_classes": 20
    }
  ],
  "usage": {
    "prompt_tokens": 3,
    "total_tokens": 3,
    "completion_tokens": 0,
    "prompt_tokens_details": null
  }
}
```

When trying other the model from the docs `jason9693/Qwen2.5-1.5B-apeach`, which has the same task of `SequenceClassification` as `papluca/xlm-roberta-base-language-detection`, seems to be returning the softmax output properly.

Seems like the same happens for offline inference, and it not just an online inference problem.

### PR DESCRIPTION
**Pull request by maxdebayser**
FIX https://github.com/vllm-project/vllm/issues/18727

By default the Bert models for sequence classification were assuming that the task was `score` and applying the same logic as the `CrossEncoder` module of the `sentence-transformers` library. In particular, this meant applying the `Identity` function at the end instead of the `Softmax` function. With this PR the pooler implementation is selected according to the task.

With these changes, the example request of issue https://github.com/vllm-project/vllm/issues/18727 returns the correct result with softmax applied:
```
curl "http://127.0.0.1:8098/classify"   -H "Content-Type: application/json"   -d '{
    "model": "papluca/xlm-roberta-base-language-detection",
    "input": "Hello"
  }'
{"id":"classify-c225e8d0d5054d72874a6340247167d3","object":"list","created":1748445777,"model":"papluca/xlm-roberta-base-language-detection","data":[{"index":0,"label":"en","probs":[0.005031585693359375,0.0194091796875,0.0047149658203125,0.005458831787109375,0.01036834716796875,0.012939453125,0.0068359375,0.0207977294921875,0.009002685546875,0.05029296875,0.010345458984375,0.0284576416015625,0.013916015625,0.712890625,0.0086517333984375,0.01119232177734375,0.01904296875,0.01247406005859375,0.0296783447265625,0.00838470458984375],"num_classes":20}],"usage":{"prompt_tokens":3,"total_tokens":3,"completion_tokens":0,"prompt_tokens_details":null}}```

Additionally, the `get_cross_encoder_activation_function` was updated to replicate updates in the `sentence-transformers`library.

cc: @noooop , @DarkLight1337

### CODE REVIEW

File: vllm/model_executor/layers/pooler.py [hunk 1]
```diff
@@ -283,30 +283,38 @@ def from_config_with_defaults(
...
+    3. Applies an activation function to the output. In the case of
+       classification models it is either sigmoid or softmax. In the
+       case of scoring models, the same behavior is configuration
+       dependent, as in the sentence-transformers library.
     """
 
     def __init__(
         self,
+        task: str,
         config: PretrainedConfig,
```
Comment (DarkLight1337) [inline vllm/model_executor/layers/pooler.py:301]: Perhaps it's better to simply pass the whole `ModelConfig` since that includes the `hf_config`

File: vllm/model_executor/layers/pooler.py [hunk 2]
```diff
@@ -283,30 +282,37 @@ def from_config_with_defaults(
...
         self,
-        config: PretrainedConfig,
+        config: ModelConfig,
         classifier: nn.Module,
         pooler: Optional[nn.Module] = None,
     ):
         super().__init__()
+        if config.task not in ["classify", "score"]:
+            raise NotImplementedError(f"task {config.task} is not supported"
+                                      " with the classification pooler")
```
Comment (DarkLight1337) [inline vllm/model_executor/layers/pooler.py:306]: Put this in the below `else` block and change the existing `else` into `elif config.task == "classify"`

File: vllm/model_executor/layers/pooler.py [hunk 3]
```diff
@@ -283,30 +282,37 @@ def from_config_with_defaults(
...
-            get_cross_encoder_activation_function(config)
+
+        if config.task == "score":
+            self.default_activation_function = \
+                get_cross_encoder_activation_function(config.hf_config)
+        elif config.task == "classify":
+            self.default_activation_function = nn.Sigmoid() \
+                if config.hf_config.num_labels == 1 else nn.Softmax()
+        else:
+            raise NotImplementedError(f"task {config.task} is not supported"
```
Comment (DarkLight1337) [inline vllm/model_executor/layers/pooler.py:314]:
```suggestion
            raise NotImplementedError(f"task={config.task!r} is not supported"
```

**Review by DarkLight1337 — APPROVED** [review]
Small nit, otherwise LGTM
